pandas/core/indexes/base.py

```diff
@@ -504,7 +504,8 @@ def __new__(
 
         elif is_ea_or_datetimelike_dtype(dtype):
             # non-EA dtype indexes have special casting logic, so we punt here
-            pass
+            if isinstance(data, (set, frozenset)):
+                data = list(data)
```

Comment:
This is related to what you mentioned on slack, I assume?

```
idx = pd.Index({1, 2}, dtype="Int64")
idx = pd.Index({1, 2}, dtype="int64")
```

Whether we consider a `set` as sequence as valid input for constructors or not (IMO it would also be fine to not do that more generally)

Reply:
Yes, correct. It#s explicitly allowed for Index, so we shouldn't break this now

Reply:
I see, for `Series(..)` we disallow it explicitly, but for `Index(..)` it currently works for non-EA dtypes. Sounds good to allow that for now.